pages/agent/v3/hooks.md

```diff
@@ -137,29 +136,25 @@ they are run as part of each job:
 
 ### Creating job lifecycle hooks
 
-Job lifecycle hooks are sourced for every job an agent accepts.
-Use job lifecycle hooks to prepare for jobs, override default behavior, or clean up after jobs that have finished.
-For example, use the `environment` hook to set a job's environment variables or the `pre-exit` hook to delete temporary files and remove containers.
-If your hook is related to the startup or shutdown of the agent, consider [agent lifecycle hooks](#job-lifecycle-hooks) for those tasks instead.
+Job lifecycle hooks are sourced for every job an agent accepts.  Use job lifecycle hooks to prepare for jobs, override default behavior, or clean up after jobs that have finished.  For example, use the `environment` hook to set a job's environment variables or the `pre-exit` hook to delete temporary files and remove containers.  If your hook is related to the startup or shutdown of the agent, consider [agent lifecycle hooks](#job-lifecycle-hooks) for those tasks instead.
 
 Job lifecycle hooks have access to all the standard [Buildkite environment variables](/docs/pipelines/environment-variables).
 
-Job lifecycle hooks are copied to `$TMPDIR` directory and *sourced* by the agent's default shell.
-This has a few implications:
+Job lifecycle hooks are copied to `$TMPDIR` directory and *sourced* by the agent's default shell. This has a few implications:
+
+>📝 A note on sourcing for shellscripting nerds
+>
+> We use the word "sourcing" in this article, but it's not strictly speaking true; the agent never actually calls the `source` command on a script. Instead, the agent uses a process called ["the scriptwrapper"](https://github.com/buildkite/agent/blob/1a5f05029cc363a984188c441f938dd316dedd16/hook/scriptwrapper.go). This process notes down the environment variables prior to a hook run, runs that hook, and then compares the environment variables after the hook run to the environment variables before the hook run. Any environment variables that were added, changed or removed are then exported to the subsequent phases and hooks. Functionally, this is very similar to how `source` works, but it's not quite the same, and it means that if you're relying on some very specific pieces of shellscripting arcana, you might find that things don't work quite as you expect.
```

Comment:
Me: should I be an absolute pedant about this?
Also me: nah
Also me: yeah
Me: ...
Me: ah fuck

The scriptwrapper runs the hook with `.` which is a synonym for `source`